src/relations/mysql_provider.py

```diff
@@ -309,7 +317,38 @@ def _on_database_broken(self, event: RelationBrokenEvent) -> None:
         relation_id = event.relation.id
         try:
             self.charm._mysql.delete_users_for_relation(relation_id)
-            logger.info(f"Removed user for relation {relation_id}")
+            logger.info(f"Removed user(s) for relation {relation_id}")
         except MySQLDeleteUsersForRelationError:
-            logger.error(f"Failed to delete user for relation {relation_id}")
+            logger.error(f"Failed to delete user(s) for relation {relation_id}")
+            return
+
+    def _on_database_provides_relation_departed(self, event: RelationDepartedEvent) -> None:
+        """Remove MySQL Router cluster metadata & router user for departing unit."""
+        if not self.charm.unit.is_leader():
+            return
+        if event.departing_unit.app.name == self.charm.app.name:
+            return
+
+        users = self.charm._mysql.get_mysql_router_users_for_unit(
+            relation_id=event.relation.id, mysql_router_unit_name=event.departing_unit.name
+        )
+        if not users:
+            return
+
+        if len(users) > 1:
+            logger.error(
+                f"More than one router user for departing unit {event.departing_unit.name}"
+            )
```

Comment:
isnt it possible to have more than 1 user returned (if there are multiple applications related to the mysqlrouter application)?